metricbeat/module/vsphere/host/host.go

```diff
@@ -111,36 +111,41 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 	}
 
 	for _, hs := range hst {
-		totalCPU := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
-		freeCPU := int64(totalCPU) - int64(hs.Summary.QuickStats.OverallCpuUsage)
-		freeMemory := int64(hs.Summary.Hardware.MemorySize) - (int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024)
-
-		event := common.MapStr{
-			"name": hs.Summary.Config.Name,
-			"cpu": common.MapStr{
-				"used": common.MapStr{
-					"mhz": hs.Summary.QuickStats.OverallCpuUsage,
-				},
-				"total": common.MapStr{
-					"mhz": totalCPU,
-				},
-				"free": common.MapStr{
-					"mhz": freeCPU,
-				},
+		cpu := common.MapStr{
+			"used": common.MapStr{
+				"mhz": hs.Summary.QuickStats.OverallCpuUsage,
 			},
-			"memory": common.MapStr{
-				"used": common.MapStr{
-					"bytes": (int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024),
-				},
-				"total": common.MapStr{
-					"bytes": hs.Summary.Hardware.MemorySize,
-				},
-				"free": common.MapStr{
-					"bytes": freeMemory,
-				},
+		}
+		memory := common.MapStr{
+			"used": common.MapStr{
+				"bytes": int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024,
 			},
 		}
 
+		if hs.Summary.Hardware != nil {
+			totalCPU := int64(hs.Summary.Hardware.CpuMhz) * int64(hs.Summary.Hardware.NumCpuCores)
+			cpu["total"] = common.MapStr{
+				"mhz": totalCPU,
+			}
+			cpu["free"] = common.MapStr{
+				"mhz": int64(totalCPU) - int64(hs.Summary.QuickStats.OverallCpuUsage),
+			}
+			memory["free"] = common.MapStr{
+				"bytes": int64(hs.Summary.Hardware.MemorySize) - (int64(hs.Summary.QuickStats.OverallMemoryUsage) * 1024 * 1024),
+			}
+			memory["total"] = common.MapStr{
+				"bytes": hs.Summary.Hardware.MemorySize,
+			}
+		} else {
+			logp.Debug("'Hardware' or 'Summary' data not found. This is either a parsing error from vsphere library, an error trying to reach host/guest or incomplete information returned from host/guest. Setting 'cpu.total.mhz', 'cpu.free.mhz', 'memory.free.bytes' and 'memory.total.bytes' to 0", "")
```

Comment:
With `log.Debug` the first entry should here be probably `vshpere`

Are the values really set to 0?

Reply:
Not now. If the values aren't found, the fields aren't being written.

Reply:
logp.Debug( has a selector as the first argument. That is what I meant above.

Reply:
If it is not written, the Debug log message should be updated.

Reply:
I have seen that in some modules (I think MySQL) the selector wasn't written but in Kafka it was in this form `var logger = logp.NewLogger("kafka")` so I have use the latter and updated all log messages to use this form.

Reply:
++ on using the new logging methods. The old ones are still very common but deprecated.